src/EventTickets/Routes/DeleteEventTicketType.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+namespace Give\EventTickets\Routes;
+
+use Give\API\RestRoute;
+use Give\EventTickets\Models\EventTicketType;
+use Give\Framework\Exceptions\Primitives\Exception;
+use WP_REST_Request;
+use WP_REST_Response;
+use WP_REST_Server;
+
+/**
+ * @unreleased
+ */
+class DeleteEventTicketType implements RestRoute
+{
+    /** @var string */
+    protected $endpoint = 'events-tickets/ticket-type/(?P<ticket_type_id>\d+)';
+
+    /**
+     * @inheritDoc
+     */
+    public function registerRoute()
+    {
+        register_rest_route(
+            'give-api/v2',
+            $this->endpoint,
+            [
+                [
+                    'methods' => WP_REST_Server::DELETABLE,
+                    'callback' => [$this, 'handleRequest'],
+                    'permission_callback' => '__return_true',
```

Comment:
A destructive action should have a permission check.

Reply:
Resolved: https://github.com/impress-org/givewp/pull/7293/commits/71a2e2ca0ff03c84aa9ac3b283fe8d56dd798a01